src/features/signUp/SignUpEmail.tsx

```diff
@@ -23,36 +26,82 @@ const SignUpButton = styled.button`
   font-size: 24px;
   font-weight: medium;
   border: none;
-`;
+  cursor: pointer;
 
-const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
-  const formData = new FormData(e.target as HTMLFormElement);
-  const data = Object.fromEntries(formData);
-
-  try {
-    const response = await apiClient.post(API_ENDPOINTS.SIGNUP_EMAIL, {
-      full_name: data.fullName,
-      email: data.email,
-      phone_number: data.phoneNumber,
-      password_hash: data.password,
-    });
-    if (response.status === 200) {
-      // 홈으로 이동
-      console.log('Sign up successful:', response.data);
-    }
-  } catch (error) {
-    console.error('Sign up failed:', error);
+  &:disabled {
+    background-color: #ccc;
+    cursor: not-allowed;
   }
-};
+`;
 
 export default function SignUpEmail() {
+  const [showErrorModal, setShowErrorModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+  const [isFullNameValid, setIsFullNameValid] = useState(false);
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [isPhoneNumberValid, setIsPhoneNumberValid] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+  const [emailError, setEmailError] = useState('');
+
+  const handleSignUp = async (e: React.FormEvent<HTMLFormElement>) => {
```

Comment:
```suggestion
  const handleSignUp = async (event: React.FormEvent<HTMLFormElement>) => {
```
나중에 검색의 용이성&가독성을 위해 event로 명명하면 어떨까요

Reply:
함수명은 폼 제출이니 handleSubmit은 어떨까요?